contracts/colony/ColonyRoles.sol

```diff
@@ -104,20 +104,25 @@ contract ColonyRoles is ColonyStorage, ContractRecoveryDataTypes {
         setTo = uint256(roles) % 2 == 1;
 
         ColonyAuthority(address(authority)).setUserRole(_user, _domainId, roleId, setTo);
-        if (roleId == uint8(ColonyRole.Recovery)) {
-          if (setTo){
-            recoveryRolesCount++;
-          } else {
-            recoveryRolesCount--;
-          }
-          emit RecoveryRoleSet(_user, setTo);
-        }
         emit ColonyRoleSet(msgSender(), _user, _domainId, roleId, setTo);
 
       }
       roles >>= 1;
       rolesChanged >>= 1;
     }
+
+    // Update the storage slot tracking number of recovery roles after all the external calls are complete
```

Comment:
Shouldn't this be before the external calls?

Reply:
Yup, I really messed up there :disappointed: